app/models/service.rb

```diff
@@ -1,10 +1,12 @@
+require 'ancestry'
+
 class Service < ApplicationRecord
   DEFAULT_PROCESS_DELAY_BETWEEN_GROUPS = 120
 
+  has_ancestry :orphan_strategy => :destroy
```

Comment:
👍 Looks good.  The top-level service is the only one that is visible so the sub-services need to be removed when it is deleted.